sktree/stats/forestht.py

```diff
@@ -1307,6 +1307,10 @@ def build_coleman_forest(
         rng.shuffle(temp_col)
         X_null[:, covariate_index] = temp_col
 
+    if not isinstance(perm_est, PermutationHonestForestClassifier):
```

Comment:
This error needs to be removed because `PermutationHonestForestClassifier` is no longer used anywhere. And the tutorials use `HonestForestClassifier` directly, so this error raising would cause problems.

Reply:
Besides simplicity, is there any reason to condense the two that I missed?

The reason the separate class is nice is for the future work, if we want to permute in more exotic ways. E.g. permute per tree, permute nearest-nbrs, etc. If the work all finishes, and we end up not needing that separate class, we can always revert to what we have here.

Reply:
Are lines 1299-1308 necessary if we use the separate class?